### PR TITLE
Add a clone_foreman_infra function

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/foremanNightlyTest.groovy [hunk 1]
```diff
@@ -10,9 +10,7 @@ pipeline {
         }
         stage('Provision Node') {
             steps {
-                dir('foreman-infra') {
-                    git url: 'https://github.com/theforeman/foreman-infra.git'
-                }
+                clone_foreman_infra(dir: 'foreman-infra')
```
Comment (komidore64) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/foremanNightlyTest.groovy]: I feel like this is more obscure than what was originally there.

**Comment by komidore64** [comment]
Where are you defining this method?

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/foremanNightlyTest.groovy (hunk 1 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/foremanNightlyTest.groovy]: `git_clone_foreman_infra()` seems clearer though

**Comment by ehelms** [comment]
Name changed and file added that has the function in it.

**Pull request merged**